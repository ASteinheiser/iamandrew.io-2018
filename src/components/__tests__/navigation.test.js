import React      from 'react';
import serializer from 'enzyme-to-json/serializer';

import Navigation from '../navigation';

expect.addSnapshotSerializer(serializer);

describe('Navigation test ->', () => {

  describe('Basic tests ->', () => {

    it('should render a navigation', () => {
      const component = shallow(
        <Navigation />
      );
      expect(component).toMatchSnapshot();
    });

    test('renders with children', () => {
      const component = mount(
        <Navigation>
          {'I\"m a child!'}
        </Navigation>
      );
      expect(component.props().children).toEqual('I\"m a child!');
      expect(component).toMatchSnapshot();
    });

    test('renders with state', () => {
      const TEST_SCREEN_WIDTH = 1024;
      const component = mount(
        <Navigation />
      );
      expect(component.state()).toEqual({ width: TEST_SCREEN_WIDTH });
      expect(component).toMatchSnapshot();
    });
  });
});
