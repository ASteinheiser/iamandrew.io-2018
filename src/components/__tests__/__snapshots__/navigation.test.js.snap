// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation test -> Basic tests -> renders with children 1`] = `
<Navigation>
  <div
    className="mac-background"
  >
    <TopNav
      width={1024}
    >
      <div
        className="top-nav-bar"
      >
        <div
          className="space-between max-width"
        >
          <div
            className="top-nav-iamandrew-logo"
            onClick={[Function]}
          >
            iamandrew
          </div>
          <div
            className="top-nav-links-container"
          >
            <div
              className="top-nav-item margin-right"
              onClick={[Function]}
            >
              projects
            </div>
            <div
              className="top-nav-item"
              onClick={[Function]}
            >
              about
            </div>
          </div>
        </div>
      </div>
    </TopNav>
    <div
      className="navigation-content max-width"
    >
      I"m a child!
    </div>
  </div>
</Navigation>
`;

exports[`Navigation test -> Basic tests -> renders with state 1`] = `
<Navigation>
  <div
    className="mac-background"
  >
    <TopNav
      width={1024}
    >
      <div
        className="top-nav-bar"
      >
        <div
          className="space-between max-width"
        >
          <div
            className="top-nav-iamandrew-logo"
            onClick={[Function]}
          >
            iamandrew
          </div>
          <div
            className="top-nav-links-container"
          >
            <div
              className="top-nav-item margin-right"
              onClick={[Function]}
            >
              projects
            </div>
            <div
              className="top-nav-item"
              onClick={[Function]}
            >
              about
            </div>
          </div>
        </div>
      </div>
    </TopNav>
    <div
      className="navigation-content max-width"
    />
  </div>
</Navigation>
`;

exports[`Navigation test -> Basic tests -> should render a navigation 1`] = `
<div
  className="mac-background"
>
  <TopNav
    width={1024}
  />
  <div
    className="navigation-content max-width"
  />
</div>
`;
